packages/driver/src/cy/commands/actions/type.ts

```diff
@@ -360,7 +361,8 @@ export default function (Commands, Cypress, cy, state, config) {
               // it sends a click event automatically if the element is a <button>,
               // but it does not if the element is an <input>.
               // event.target is null when used with shadow DOM.
-              (!isFirefoxBefore98 && event.target && $elements.isInput(event.target))
+              (!isFirefoxBefore98 && event.target && $elements.isInput(event.target)) ||
+              (isFirefox106OrLater && $elements.isButton(event.target))
```

Comment:
```suggestion
              (!isFirefoxBefore98 && event.target && $elements.isInput(event.target)) ||
              (!isFirefoxBefore91 && event.target && $elements.isButton(event.target))
```

This is what my testing yielded for simulated space `keyup` events on buttons:

Firefox < 91: Firefox fires the click
Firefox >= 91: Firefox __does not__ fire the click

Adding the `event.target` check will also fix the failing driver tests.

Reply:
OK cool I will get fixed, update the tests, and will mark as a bugfix in the changelog

Reply:
I think I have fixed in 9a1326bc8c. should be ready for re review